python/pyarrow/_orc.pyx

```diff
@@ -403,6 +406,7 @@ cdef class ORCWriter(_Weakrefable):
         cdef:
             shared_ptr[WriteOptions] write_options
         self.sink = sink
+        self.close_file = close_file
```

Comment:
If `sink` is not a file object but e.g. a file path, the file should always be closed at the end, however.

Reply:
See my comment above about how the parquet cython code handles this:

https://github.com/apache/arrow/blob/daa5c18e9697a6455a7a75fec19594543c17b21e/python/pyarrow/_parquet.pyx#L1450-L1459

and 

https://github.com/apache/arrow/blob/daa5c18e9697a6455a7a75fec19594543c17b21e/python/pyarrow/_parquet.pyx#L1486-L1491

That also avoids that this keyword needs to be propagated to the public API (as Antoine mentions, the fixed default of False (or True) would not be correct in all cases)